File: src/test/resources/jython/mean2d.py
# CLIJ example ImageJ Ops jython: mean.py
#
# This script shows how the mean average filter works in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

mean = ops.run("CLIJ_mean2D", inputGPU, 3);
# show result
result = ops.run("CLIJ_pull", mean)
ui.show("mean2D", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", mean)
ops.run("CLIJ_close")

File: src/test/java/net/haesleinhuepf/clij/ops/examples/Mean2D.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_meanBox.CLIJ_meanBox;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class Mean2D {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif")
			.getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);

		Object mean = ij.op().run(CLIJ_meanBox.class, inputGPU, 3, 3);

		Object result = ij.op().run(CLIJ_pull.class, mean);
		ij.ui().show("mean2D", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, mean);


	}

	public static void main(String... args) throws IOException {
		Mean2D task = new Mean2D();
		task.run();
	}
}
